docs/source/getting-started/what-is-jupyterhub.md

```diff
@@ -0,0 +1,465 @@
+allow you to run the kernels on a different machine and possibly with
+a different environment.
+
+A kernel doesn't just execute it's language - cell magics such as `%`,
+`%%`, and `!` are a property of the kernel - in particular, these are
+IPython kernel commands and don't necessarily work in any other
+kernel unless they specifically support them.
+
+What does this mean?  There is yet *another* layer of configurability.
+Each kernel can run a different programming language, with different
+software, and so on.  By default, they would run in the same
+environment as the single-user notebook server, and the most common
+other way they are configured is by
+running in different Python virtual environments or conda
+environments.  They can be started and killed independently (there is
+normally one per notebook you have open).  The kernels is what uses
+most of your memory and CPU when running Jupyter - the rest of the web
+interface has a small footprint.
+
+You can list your installed kernels with `jupyter kernelspec list`.
+If you look at one of `kernel.json` files in those directories, you
+will see exactly what command is run.  These are normally
+automatically made by the kernels, but can be edited as needed.  [The
+spec](https://jupyter-client.readthedocs.io/en/stable/kernels.html)
+tells you even more.
+
+The normally has to be reachable by the single-user notebook server
+but the gateways mentioned above can get around that limitation.
+
+If you get problems with "Kernel died" or some other error in a single
+notebook but the single-user notebook server stays working, it is
+usually a problem with the kernel.  It could be that you are trying to
+use more resources than you are allowed and the symptom is the kernel
+getting killed.  It could be that it crashes for some other reason.
+In these cases, you need to find the kernel logs and investigate.
+
+The debug logs for the kernel are normally mixed in with the
+single-user notebook server logs.
+
+
+
+## JupyterHub distributions
+
+There are several "distributions" which automatically install all of
+the things above and configure them for a certain purpose.  They are
+good ways to get started, but if you have custom needs, eventually it
+may become hard to adapt them to your requirements.
+
+* [**Zero to JupyterHub with
+  Kubernetes**](https://zero-to-jupyterhub.readthedocs.io/) installs
+  an entire scaleable system using Kubernetes.  Uses KubeSpawner,
+  ....Authenticator, ....
+
+* [**The Littlest JupyterHub**](https://tljh.jupyter.org/) installs JupyterHub on a single system
+  using SystemdSpawner and NativeAuthenticator (which manages users
+  itself).
+
+* [**JupyterHub the hard
+  way**](https://jupyterhub.readthedocs.io/en/stable/installation-guide-hard.html)
+  takes you through everything yourself.  It is a natural companion to
+  this guide, since you get to experience every little bit.
+
+
+
+## I want to...
+
+TODO: answers to common cross-layer questions.
```

Comment:
I think the foregoing text actually does a good job of laying things out.  You might want to omit this section for now to move forward with getting these docs integrated and then add this section later if things come up that aren't handled better any other way.